admission/webhook/validator.go

```diff
@@ -0,0 +1,187 @@
+package webhook
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"slices"
+
+	"github.com/kubescape/go-logger"
+	"github.com/kubescape/go-logger/helpers"
+	"github.com/kubescape/k8s-interface/k8sinterface"
+	exporters "github.com/kubescape/operator/admission/exporter"
+	v1 "k8s.io/api/core/v1"
+	rbac "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/admission"
+)
+
+// TODO: Implement the validations in a separate package.
+
+type AdmissionValidator struct {
+	kubernetesClient *k8sinterface.KubernetesApi
+	exporter         *exporters.HTTPExporter
+}
+
+func NewAdmissionValidator(kubernetesClient *k8sinterface.KubernetesApi, exporter *exporters.HTTPExporter) *AdmissionValidator {
+	return &AdmissionValidator{
+		kubernetesClient: kubernetesClient,
+		exporter:         exporter,
+	}
+}
+
+// We are implementing the Validate method from the ValidationInterface interface.
+func (av *AdmissionValidator) Validate(ctx context.Context, attrs admission.Attributes, o admission.ObjectInterfaces) (err error) {
+	kind := attrs.GetKind().GroupKind().Kind
+	resource := attrs.GetResource().Resource
+
+	switch {
+	case kind == "Pod" || resource == "pods":
+		// If the request is for a pod, we call the validatePods function to validate the request.
+		return av.validatePods(attrs)
+	case kind == "ClusterRoleBinding" || resource == "clusterrolebindings":
+		// If the request is for a clusterRoleBinding, we call the validateClusterRoleBinding function to validate the request.
+		return av.validateAdminClusterRoleBinding(attrs)
+	case kind == "RoleBinding" || resource == "rolebindings":
+		// If the request is for a roleBinding, we call the validateRoleBinding function to validate the request.
+		return av.validateAdminRoleBinding(attrs)
+	default:
+		return nil
+	}
+}
+
+// We are implementing the Handles method from the ValidationInterface interface.
+// This method returns true if this admission controller can handle the given operation, we accept all operations.
+func (av *AdmissionValidator) Handles(operation admission.Operation) bool {
+	return true
+}
+
+func (av *AdmissionValidator) validateAdminRoleBinding(attrs admission.Attributes) error {
```

Comment:
I think at some point these will reside as separate rules, such as in node-agent?

Reply:
Maybe at some point but no now.